clash-prelude/src/Clash/Sized/Internal/Unsigned.hs

```diff
@@ -308,7 +308,7 @@ instance KnownNat n => Enum (Unsigned n) where
 
   pred n
     | n == minBound =
-        error $ "'pred' was called on (" <> show @(Unsigned n) maxBound <> " :: "
+        error $ "'pred' was called on (" <> show @(Unsigned n) minBound <> " :: "
              <> "Unsigned " <> show (natToNatural @n) <> ") and caused an "
              <> "underflow. Use 'satPred' and specify a SaturationMode if you "
```

Comment:
While we're changing the error message anyway, how about:
```suggestion
             <> "overflow. Use 'satPred' and specify a SaturationMode if you "
```
In my book, an underflow is something that happens to representations of fractional numbers, not whole numbers.